api/ws-server.js

```diff
@@ -1,49 +1,27 @@
-const WebSocket = require('ws')
+const Server = require('socket.io')
 
 const logger = require('./config/winston')
 const { appEmitterBus } = require('./emitters.js')
-const { serialize, unserialize } = require('./utils/helpers.js')
-
-const startPing = ({ wss, interval = 30 * 1000 }) =>
-  setInterval(() => {
-    wss.clients.forEach((ws) => {
-      if (!ws.isAlive) {
-        return ws.terminate()
-      }
-      ws.isAlive = false
-      ws.send(serialize({ message: 'ping' }))
-    })
-  }, interval)
+const { serialize } = require('./utils/helpers.js')
 
 const setupWss = async (server, sessionMiddleware) => {
-  const wss = new WebSocket.Server({
-    server,
-    verifyClient: ({ req }, done) => {
-      return sessionMiddleware(req, {}, () => done(req.session.id))
-    }
+  const io = new Server(server)
+
+  io.use((socket, next) => {
+    sessionMiddleware(socket.request, socket.request.res, next)
   })
 
-  wss.on('connection', (ws, req) => {
-    logger.info('Web client connected.')
-    ws.req = req
-    ws.isAlive = true
+  io.origins([process.env.APP_URL])
 
-    ws.on('pong', () => (ws.isAlive = true))
-    ws.on('message', (msg) => {
-      msg = unserialize(msg)
-      if (msg.message === 'pong') {
-        ws.isAlive = true
-      }
-    }) // silent client message
+  io.on('connection', (ws) => {
+    logger.info('Web client connected.')
 
     appEmitterBus.on('solution-judgement-available', (msg) => {
-      if (ws.readyState === WebSocket.OPEN && msg) {
-        ws.send(serialize({ message: 'solution', judgement: msg }))
+      if (msg) {
+        ws.emit('solution-judgement-available', serialize({ message: 'solution', judgement: msg }))
```

Comment:
Do we need to serialize even? `io` should handle objects as they are right away.

Reply:
Oh I see. Will change it tomorrow.